src/index_spec.js

```diff
@@ -0,0 +1,134 @@
+const sinon = require('sinon');
+const Uploader = require('./index.js');
+const FileAPI = require('file-api')
+const {expect} = require('chai');
+const {describe} = require('mocha');
+
+global.FormData = require('form-data');
+global.XMLHttpRequest = sinon.useFakeXMLHttpRequest();
+
+describe('Uploader', () => {
+    let uploader, file, fileList;
+
+    beforeEach(() => {
+        file = new FileAPI.File({ 
+            name: "abc-song.txt",
+            type: "text/plain",
+            buffer: new Buffer("abcdefg,hijklmnop, qrs, tuv, double-u, x, y and z")
+          });
+        fileList = new FileAPI.FileList([file]);
+        uploader = new Uploader('https://fiverr.com', {});        
+        
+    })
+
+    describe('File names', () => {
+        it('should strip name from extention', () => {
+            expect(Uploader.stripNameFromExtension('test.jpg')).to.deep.equal({ name: 'test', extension: '.jpg' });
+        });
+        
+        it('should return the last extention', () => {
+            expect(Uploader.stripNameFromExtension('test.jpg.img')).to.deep.equal({ name: 'test.jpg', extension: '.img' });
+        });
+
+        it('should return empty space when there is not extention', () => {
+            expect(Uploader.stripNameFromExtension('test')).to.deep.equal({ name: 'test', extension: '' });
+        });
+    });
+
+    describe('File Id', () => {
+        it('should start with \'file_\'', () => {
+            expect(Uploader.itemId('test.jpg').startsWith('file_')).to.equal(true);
+        });
+        
+        it('should allow _', () => {
+            expect(Uploader.itemId('test_.jpg').startsWith('file_test_jpg')).to.equal(true);            
+        });
+
+        it('should allow numbers', () => {
+            expect(Uploader.itemId('1.jpg').startsWith('file_1jpg')).to.equal(true);            
+        });
+
+        it('should ignore special characters', () => {
+            expect(Uploader.itemId('!!1@#$@#!$.jpg').startsWith('file_1jpg')).to.equal(true);            
+        });
+    });
+
+    describe('Pretty File Size', () => {
+        it('Default value should be 0KB', () => {
+            expect(Uploader.prettyFileSize()).to.equal('0KB');
```

Comment:
expect(...).to.throw(...)

Imo. If the default is just for emulated files, it should be made an error instead.

Reply:
lets talk about it offline